### DISCUSSION TITLE
Will scalene work without GIL (such as JPython)?

### DISCUSSION THREAD

**Discussion by StarDuster** [plasma-umass/scalene#248] [Q&A] [answered]
I studied Scalene's paper, and in the section "Managing Signals", it described that some behavior relied on the GIL (which ensures no concurrency inside the interpreter). So in my mind, Scalene shouldn't be used on interpreters without GIL, such as JPython? I'm not sure whether I'm following the paper right.

**Comment by emeryberger** [comment, marked as answer]
Scalene leverages several aspects of the CPython interpreter. We've restructured some aspects of Scalene so it (incidentally, probably) isn't so dependent on the presence of the GIL. That said, we have not looked into JPython or other Python implementations (Scalene used to work with `pypy`, but no longer does due to some missing features).
